src/schedule/ll_schedule.c

```diff
@@ -286,14 +286,6 @@ static int schedule_ll_domain_set(struct ll_schedule_data *sch,
 		task->start = domain->next_tick;
 	} else {
 		/* later periodic task, simplify and cover it by the coming interrupt */
-#ifdef __ZEPHYR__
```

Comment:
Yes, I think we should not MIX these two worlds. We really need a layer of abstraction so if ZEPHYR scheduler is selected lets use it as it is.

Reply:
@mrajwa yep, that the plan - we get the Zephyr code to work with existing external scheduling APIs and stable then we adapt the externals to match the features required and features exported by Zephyr.